codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/endpoints/EndpointParametersGenerator.java

```diff
@@ -0,0 +1,299 @@
+                                    return $fmtErrorf:T("parameter $memberName:L is required")
+                                }
+                                """,
+                                commonCodegenArgs,
+                                MapUtils.of(
+                                        "memberName", getExportedParameterName(parameter)));
+                    });
+                    w.write("return nil");
+                });
+
+    }
+
+    public static Symbol parameterAsSymbol(Parameter parameter) {
+        return switch (parameter.getType()) {
+            case STRING -> SymbolUtils.createPointableSymbolBuilder("string")
+                    .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true).build();
+
+            case BOOLEAN -> SymbolUtils.createPointableSymbolBuilder("bool")
+                    .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true).build();
+        };
+    }
+
+    public static boolean haveRequiredParameters(Parameters parameters) {
+        for (Parameter parameter : parameters.toList()) {
+            if (parameter.isRequired()) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static String getExportedParameterName(Parameter parameter) {
+        return StringUtils.capitalize(parameter.getName().asString());
+    }
+
+    public static String getExportedParameterName(StringNode name) {
+        return StringUtils.capitalize(name.getValue());
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder implements SmithyBuilder<EndpointParametersGenerator> {
+        private Symbol parametersType;
+
+        private Builder() {
+        }
+
+        public Builder parametersType(Symbol parametersType) {
+            this.parametersType = parametersType;
+            return this;
+        }
+
+        @Override
+        public EndpointParametersGenerator build() {
+            return new EndpointParametersGenerator(this);
+        }
+    }
+
+    public static Stream<Parameter> sortParameters(Parameters parameters) {
+        return parameters.toList().stream().sorted(new Sorted());
+    }
+
+    public static final class Sorted implements Comparator<Parameter>, Serializable {
+        /**
+         * Initializes the SortedMembers.
+         */
+        @Override
+        public int compare(Parameter a, Parameter b) {
+            // first compare if the options are required or not, whichever option is
+            // required should win. If both
+            // options are required or not required, continue on to alphabetic search.
+
+            // If a is required but b isn't return -1 so a is sorted before b
+            // If b is required but a isn't, return 1 so a is sorted after b
+            // If both a and b are required or optional, use alphabetic sorting of a and b's
+            // member name.
+
+            int requiredOption = 0;
```

Comment:
yah i thought it might be useful for potential issues with changed member order breaking ABI compatibility. But I was informed that changed member order doesnt effect Go because Go statically links everything.

Reply:
ill remove it.